tests/WindowingTests.cpp

```diff
@@ -0,0 +1,50 @@
+#include "common.h"
+#include <wil/windowing.h>
+
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && WIL_HAS_CXX_17
+TEST_CASE("EnumWindows", "[windowing]")
+{
+    // lambda can return a bool
+    wil::for_each_window([](HWND hwnd)
+        {
+            REQUIRE(IsWindow(hwnd));
+            return true;
+        });
+
+    // or not return anything (we'll stop if we get an exception)
+    wil::for_each_window([](HWND hwnd)
+        {
+            REQUIRE(IsWindow(hwnd));
+        });
+
+    // With captures
+    const auto pid = GetCurrentProcessId();
+    wil::for_each_window([pid](HWND hwnd)
+        {
+            if (pid == GetWindowThreadProcessId(hwnd, nullptr))
+            {
+                REQUIRE(IsWindow(hwnd));
+            };
+            return true;
+        });
+
+    const auto thread_id = GetCurrentThreadId();
+    wil::for_each_thread_window(thread_id, [](HWND hwnd)

```

Comment:
Question: when running the tests, does the callback get invoked at all? Maybe the console widow, but I'm somewhat doubtful of that. If not, might be good to use the thread id of something that's guaranteed to have windows (like Explorer's desktop maybe?)